garak/generators/huggingface.py

```diff
@@ -505,18 +554,24 @@ def __init__(
                 self.name, padding_side="left"
             )
 
-        self.do_sample = do_sample
         self.generation_config = transformers.GenerationConfig.from_pretrained(
             self.name
         )
         self.generation_config.eos_token_id = self.model.config.eos_token_id
         self.generation_config.pad_token_id = self.model.config.eos_token_id
 
+    def _clear_client(self):
+        self.model = None
+        self.config = None
+        self.tokenizer = None
+        self.generation_config = None
```

Comment:
Why do we clear this? Does leaving it around cause conflicts with `_load_client()`'s application of config?

Reply:
This value is a `transformers.GenerationConfig` object and is always replace in a call to `_load_client()`.  By clearing it we ensure it can be garbage collected and avoid a dependency on `transformer` objects being able to support `pickle` if we follow the same `custom` pickle pattern from openai generators in the future.